### PR TITLE
Bench 53 refactor with typing test categories

### PR DESCRIPTION
**Pull request by PietroConvalleAD**
Adds typing in categories tests

### CODE REVIEW

File: answerking_app/tests/API_types.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class NewItemType(TypedDict):
+    name: str
+    price: str
+    description: str
+    stock: int
+    calories: int
+
+
+class ItemIDType(TypedDict):
+    id: int
```
Comment (NogaAD) [inline answerking_app/tests/API_types.py]: What is the benefit of having ItemIdType and CategoryIDType instead of just IDType which apply for both?

Comment (PietroConvalleAD) [reply answerking_app/tests/API_types.py]: Good point, will change this

File: answerking_app/tests/API_types.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    pass
+
+
+class NewCategoryType(TypedDict):
+    name: str
+    items: list[ItemType]
+
+
+class CategoryType(IDType, NewCategoryType):
+    pass
```
Comment (abbas-khan8) [inline answerking_app/tests/API_types.py:26]: This is something we should probably be discussing as a group. Have you reached out to the other guys to see if they are implementing their tests in the same way as this?

Comment (PietroConvalleAD) [reply answerking_app/tests/API_types.py:26]: I don't think anyone has started a test ticket yet, joe has added it to the views

Comment (abbas-khan8) [reply answerking_app/tests/API_types.py:26]: Cool as long as everyone who starts a testing refactor ticket is on board

File: .gitignore [hunk 3]
```diff
@@ -145,4 +145,4 @@ GitHub.sublime-settings
 *.sqlite3
 
 # Poetry
-poetry.lock
+poetry.lock
```
Comment (abbas-khan8) [inline .gitignore]: No changes needed to this file it's in develop

File: answerking_app/tests/test_categories.py [hunk 4]
```diff
@@ -273,9 +282,9 @@ def test_put_valid_without_items_returns_ok(self):
         # Arrange
         old_category = client.get(f"/api/categories/{self.test_cat_1.id}").json()
```
Comment (JoeCSykes) [inline answerking_app/tests/test_categories.py:280]: does this not need a type?

File: answerking_app/tests/test_categories.py [hunk 5]
```diff
@@ -303,8 +312,8 @@ def test_put_valid_without_items_returns_ok(self):
         )
         actual = response.json()
```
Comment (JoeCSykes) [inline answerking_app/tests/test_categories.py:310]: does this not need a type?

File: answerking_app/tests/test_categories.py (hunk 4 above)
Comment (PietroConvalleAD) [reply answerking_app/tests/test_categories.py:280]: Haven't typed those as they are quite generic and tested with the assertion, but yeah, I can add them

Comment (PietroConvalleAD) [reply answerking_app/tests/test_categories.py:280]: I actually checked the type now and I don't think it's worth the effort, they are quite messy objects


for instance:
```
def put(self,
        path: Any,
        data: str = "",
        content_type: str = "application/octet-stream",
        follow: bool = False,
        secure: bool = False,
        **extra: Any) -> WSGIRequest | {redirect_chain, status_code, url}
```

Comment (JoeCSykes) [reply answerking_app/tests/test_categories.py:280]: ah ok. I'm happy for these to be left  untyped

**Review by JoeCSykes — APPROVED** [review]

**Review by NogaAD — APPROVED** [review]

**Pull request merged**